src/axom/core/ArrayIteratorBase.hpp

```diff
@@ -48,8 +48,8 @@ class ArrayIteratorBase
    */
   ValueType& operator*()
   {
-    return (*m_arrayPtr)[IteratorBase<ArrayIteratorBase<ArrayType, ValueType>,
-                                      IndexType>::m_pos];
+    return m_arrayPtr->flatIdx(
+      IteratorBase<ArrayIteratorBase<ArrayType, ValueType>, IndexType>::m_pos);
```

Comment:
Minor/style: The code might be a bit more readable if the base type were made a type alias for the class

e.g.
```cpp
using BaseType  = IteratorBase<ArrayIteratorBase<ArrayType, ValueType>, IndexType>;
```

Then this function could be:
```cpp
    return m_arrayPtr->flatIdx(BaseType::m_pos);
``

Reply:
I support this suggestion, after much thought.